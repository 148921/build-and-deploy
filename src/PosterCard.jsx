import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';
import Chip from '@material-ui/core/Chip';
import Button from '@material-ui/core/Button';

const styles = {
  card: {
    maxWidth: 500,
  },
  media: {
    height: 750,
  },
};

const constructMovieDetailsUrl = movieId => `https://www.themoviedb.org/movie/${movieId}`;
const redirectToMoviePage = (movieId) => {
  window.location = constructMovieDetailsUrl(movieId);
};

// This component renders the movie poster and some details about the movie
const PosterCard = ({ classes, posterUrl, movieDetails }) => (
  <div>
    <Card className={classes.card}>
      <CardMedia className={classes.media} image={posterUrl} title={movieDetails.title} />
      <CardContent>
        <Typography variant="h5" component="h2">
          {movieDetails.title}
        </Typography>
        <Typography paragraph className={classes.secondaryText} color="textSecondary">
          (
          {new Date(movieDetails.release_date).getFullYear()}
          )
        </Typography>
        <Typography paragraph component="p">{movieDetails.overview}</Typography>
        <Chip label={`Rating: ${movieDetails.vote_average}`} className={classes.chip} variant="outlined" />
      </CardContent>
      <CardActions>
        <Button
          size="small"
          color="primary"
          onClick={() => redirectToMoviePage(movieDetails.id)}
        >
          Read More
        </Button>
      </CardActions>
    </Card>
  </div>
);

PosterCard.propTypes = {
  classes: PropTypes.object.isRequired,
  posterUrl: PropTypes.string.isRequired,
  movieDetails: PropTypes.object.isRequired,
};

export default withStyles(styles)(PosterCard);
