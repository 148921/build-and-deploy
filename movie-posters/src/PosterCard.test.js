import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PosterCard from "./PosterCard";

describe("PosterCard", () => {
  const defaultProps = {
    posterUrl: "/mock-poster.png",
    movieDetails: {
      title: "Mock Movie 14",
      description: "Some movie description"
    }
  };

  it("should render correctly", () => {
    const props = { ...defaultProps };
    const wrapper = shallow(<PosterCard {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
