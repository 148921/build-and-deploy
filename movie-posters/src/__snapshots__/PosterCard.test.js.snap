// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterCard should render correctly 1`] = `
<PosterCard
  classes={
    Object {
      "card": "PosterCard-card-1",
      "media": "PosterCard-media-2",
    }
  }
  movieDetails={
    Object {
      "description": "Some movie description",
      "title": "Mock Movie 14",
    }
  }
  posterUrl="/mock-poster.png"
/>
`;
